cpp/server/proxy.cc

```diff
@@ -0,0 +1,181 @@
+#include "server/proxy.h"
+
+#include <algorithm>
+#include <event2/http.h>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "monitoring/monitoring.h"
+#include "proto/ct.pb.h"
+#include "server/json_output.h"
+#include "util/libevent_wrapper.h"
+
+
+using ct::ClusterNodeState;
+using std::bind;
+using std::getline;
+using std::make_pair;
+using std::pair;
+using std::placeholders::_1;
+using std::rand;
+using std::string;
+using std::stringstream;
+using std::unique_ptr;
+using std::unordered_set;
+using std::vector;
```

Comment:
There's ```util::Executor``` and ```util::Task``` used below that could be here, but they don't really have to, only a few uses, and they're not too obtrusive, your call.